spec/features/score_spec.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe 'Score', js: true do
+  let(:exercise) { create(:hello_world) }
+  let(:user) { create(:external_user) }
+  let(:scoring_response_full_score) do
+    [{
+      status: :ok,
+      stdout: '',
+      stderr: '',
+      waiting_for_container_time: 0,
+      container_execution_time: 0,
+      file_role: 'teacher_defined_test',
+      count: 1,
+      failed: 0,
+      error_messages: [],
+      passed: 1,
+      score: 1.0,
+      filename: 'index.html_spec.rb',
+      message: 'Well done.',
+      weight: 2.0,
+    }]
+  end
+  let(:scoring_response_zero_points) do
```

Comment:
Indeed, the response below does not grant any good score. However, it's also not a real example of a failed test. The following would be, but of course is too complex. Maybe you can construct a nice simplified example, suitable for the given test?

Besides adjusting, you could also consider moving the scoring responses to a factory.

```json
{
    "file_role": "teacher_defined_test",
    "waiting_for_container_time": 0.149172,
    "stdout": "javac -encoding utf8 -cp .:/usr/java/lib/hamcrest-core-1.3.jar:/usr/java/lib/junit-4.13.jar HalloWeltTest2.java\r\njava -cp .:/usr/java/lib/hamcrest-core-1.3.jar:/usr/java/lib/junit-4.13.jar org.junit.runner.JUnitCore HalloWeltTest2\r\nJUnit version 4.13.2\r\n.E.\r\nTime: 0.004\r\nThere was 1 failure:\r\n1) checkForCorrectOutput(HalloWeltTest2)\r\norg.junit.ComparisonFailure: Es wird ein falscher String ausgegeben expected:<willkommen detektiv[ duke]!\r\n> but was:<willkommen detektiv[]!\r\n>\r\n\tat org.junit.Assert.assertEquals(Assert.java:117)\r\n\tat HalloWeltTest2.checkForCorrectOutput(HalloWeltTest2.java:37)\r\n\r\nFAILURES!!!\r\nTests run: 2,  Failures: 1\r\n\r\n",
    "stderr": "make: *** [Makefile:7: test] Error 1\r\n",
    "exit_code": 2,
    "container_execution_time": 0.509629,
    "status": "failed",
    "count": 2,
    "failed": 1,
    "error_messages": [
        "Es wird ein falscher String ausgegeben expected:<willkommen detektiv[ duke]!\r\n> but was:<willkommen detektiv[]!\r\n>\r\n"
    ],
    "passed": 1,
    "score": 0.5,
    "filename": "HalloWeltTest2.java",
    "message": "Die Ausgabe weicht von “Willkommen Detektiv Duke!” ab. Überprüfe den Inhalt von System.out.println() auf Schreibfehler\n",
    "weight": 2.5
}
```